[PHYW-537] Rest in otdoa

serranozack opened the pull request:
<!-- CURSOR_SUMMARY -->
> [!NOTE]
> Extracts a reusable HTTP API, removes legacy REST implementation, routes all messages through a single RS thread/handler, and replaces Kconfig URL with a code default.
> 
> - **Core architecture**:
>   - Unifies message processing via `otdoa_handle_message()`; HTTP messages now funneled through the RS thread (removed separate HTTP workqueue/path).
>   - Adds new public HTTP API (`include/otdoa_al/otdoa_http_api.h`) with `bind/unbind/connect/disconnect/send_request`, callbacks, and helpers.
> - **HTTP implementation**:
>   - Implements the new API in `otdoa_thread_http.c` (TLS setup, IP checks, socket ops).
>   - Simplifies `otdoa_http.c` to delegate to H1 handlers and new API; introduces `otdoa_http_send_message(void*, ...)`.
>   - Removes legacy HTTP/REST code (`otdoa_http_rest.c/.h`).
> - **Config/Defaults**:
>   - Removes `CONFIG_OTDOA_BSA_DL_SERVER_URL`; uses `DEFAULT_BSA_DL_SERVER_URL` in code.
>   - Updates docs and Kconfig accordingly; keeps `CONFIG_OTDOA_HTTPS_PORT/HTTP_PORT`, upload settings.
> - **Threading & queues**:
>   - Deletes HTTP workqueue; message slab sized by `OTDOA_HTTP_MAX_MSG_SIZE` and routes via `otdoa_queue_rs_message`.
>   - Replaces stop handling with atomic flag `http_stop_pending`.
> - **Headers/Includes**:
>   - Adds prototypes for `otdoa_http_handle_message(void*)` and `otdoa_handle_message(void*)`; updates includes to use the new HTTP API header.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 5a884bfb934e69d84efdcf6a3926873fc7c09570. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

serranozack committed:
lib: otdoa_al: remove http rest API functions

pulls out otdoa_http_rest.c to go in the otdoa_lib

Signed-off-by: Zack Mudd <zmudd@serranosystems.com>

nriedel commented on include/otdoa_al/otdoa_http_api.h:
```diff
@@ -0,0 +1,48 @@
+
+int otdoa_http_send_message(void *pMsg, uint32_t len);
+void otdoa_http_invoke_callback_dl_compl(int status);
+void otdoa_http_invoke_callback_ul_compl(int status);
```
These two functions should be in the library.  I think that means the callback is only in the library as well.

nriedel commented on lib/otdoa_al/otdoa_http_api.c:
I think we should remove this, and just have the callback in the library.

nriedel commented on lib/otdoa_al/otdoa_thread.c:
```diff
@@ -200,27 +142,15 @@ int otdoa_queue_rs_message(const void *pv_msg, const size_t length)
-		return -1;
+	if (0 != atomic_set(&gOTDOA.http_stop_pending, fail_or_cancel)) {
+		// there was a non-zero value already pending, so we've "double-stopped"
+		return OTDOA_API_INTERNAL_ERROR;
```
Maybe log a warning here, since this results from the user cancelling twice.

nriedel commented on lib/otdoa_al/otdoa_thread_http.c:
```diff
@@ -157,42 +164,43 @@ int http_bind(tOTDOA_HTTP_MEMBERS *pG, const char *pURL)
+ * @param res Pointer to addrinfo to unbind
  */
-void http_unbind(tOTDOA_HTTP_MEMBERS *pG)
+int otdoa_http_unbind(void *res)
```
I'm feeling nervous about this since we had the issue with double frees of this structure.  I added this stuff:
```
	/* always unbind first */
	otdoa_http_unbind(*res);
    *res = NULL;
```
but I think that setting the pointer to NULL should be done in the unbind, so we don't have to do it after each call to otdoa_http_unbind().  So maybe pass in a `void**` and set the point to NULL in otdoa_http_unbind()?

nriedel commented on lib/otdoa_al/otdoa_thread_http.c:
```diff
@@ -203,56 +211,61 @@ void http_unbind(tOTDOA_HTTP_MEMBERS *pG)
  * failure
  */
-int http_connect(tOTDOA_HTTP_MEMBERS *pG, const char *tls_host)
+int otdoa_http_connect(int *fdSocket, struct sockaddr *res, char *szModemAddress,
```
I'm confused.  I though struct sockaddr was the thing giving us problems with pos_linux compilation.

serranozack replied:
That was actually addrinfo, but now for a HOST build I have `otdoa_http_api.h` providing a dummy definition of both structs, since we don't actually do any network ops in the tests

nriedel reviewed (changes requested):
Looks good.  Just a few minor fixes needed.

serranozack committed:
lib: otdoa_al: move HTTP message internals to otdoa_lib

Removes the separate HTTP workqueue and handles messages on RS thread
Prefixes http API functions with otdoa_
Moves OTDOA callback handling inside otdoa_lib
Fixes conditional inclusion of results upload functionality
Provides mock network structs for testing on HOST builds

Signed-off-by: Zack Mudd <zmudd@serranosystems.com>
